src/e3/collection/dag.py

```diff
@@ -75,19 +75,18 @@ def next_element(
         result = next((k for k in self.non_visited if self.pred_number[k] == 0), None)
 
         if result is None:
+            for node in self.non_visited:
```

Comment:
when busy_state is enabled it's legal to get a None (when no nodes are ready)

Reply:
The git preview does not help, but the line 86 should ensure that if we do not have a cycle and we are in busy_state, then we can return None